atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraRequestExceptionHandler.java

```diff
@@ -278,7 +278,8 @@ public boolean shouldBackoff(Exception ex) {
 
         @Override
         public long getBackoffPeriod(int numberOfAttempts) {
-            return Math.min(500 * (long) Math.pow(2, numberOfAttempts), MAX_BACKOFF);
+            double randomCoeff = ThreadLocalRandom.current().nextDouble() + 0.5;
```

Comment:
Feels a bit odd that we specify the average maximum but don't really control the actual maximum - I think we really want to control the maximum and have that be 30.  You could either cap everything at 30s, or take 30s as the maximum, then calculate 2/3 of this and apply the +/-50% from this value.